server/lib/device/device.saveHistoricalState.js

```diff
@@ -0,0 +1,69 @@
+const db = require('../../models');
+const logger = require('../../utils/logger');
+const { BadParameters } = require('../../utils/coreErrors');
+
+/**
+ * @description Save hstorical device feature state in DB.
+ * @param {Object} deviceFeature - A DeviceFeature object.
+ * @param {Object} historicalState - The historical feature state object of the deviceFeature to save.
+ * @param {Object} createdAt - Date of state.
+ * @example
+ * saveHistoricalState({
+ *      id: 'fc235c88-b10d-4706-8b59-fef92a7119b2',
+ *      selector: 'my-light'
+ *    },
+ *    'newValue',
+ *    '2022-02-03'
+ * );
+ */
+async function saveHistoricalState(deviceFeature, historicalState, createdAt) {
+  logger.debug(`device.saveHistoricalState of deviceFeature ${deviceFeature.selector}`, historicalState);
+
+  if (Number.isNaN(historicalState)) {
+    throw new BadParameters(`device.saveHistoricalState of NaN value on ${deviceFeature.selector}`);
+  }
+  // get current feture to update last value if needed
+  const deviceFeatureInDB = await db.DeviceFeature.findOne({
```

Comment:
Why don't you take this value from the RAM state?

Like we do on the other device.setValue function ?

```
const previousDeviceFeature = this.stateManager.get('deviceFeature', deviceFeature.selector);
```

DB calls are expensive, the RAM state is made for that :)

Reply:
i fix it